python/ray/exceptions.py

```diff
@@ -829,6 +829,15 @@ class ObjectRefStreamEndOfStreamError(RayError):
     pass
 
 
+@DeveloperAPI
+class OufOfBandRefSerializationException(RayError):
+    """Raised when an `ray.ObjectRef` is out of band serialized by
+    `ray.cloudpickle`. It is an anti pattern.
```

Comment:
link to the doc?

Reply:
I can't yet (because doc is not there yet). we should do in a follow up